components/script/dom/htmlmediaelement.rs

```diff
@@ -1279,12 +1282,32 @@ impl HTMLMediaElementMethods for HTMLMediaElement {
     // https://html.spec.whatwg.org/multipage/#dom-media-autoplay
     make_bool_setter!(SetAutoplay, "autoplay");
 
+    // https://html.spec.whatwg.org/multipage/#dom-media-defaultmuted
+    make_bool_getter!(DefaultMuted, "muted");
+    // https://html.spec.whatwg.org/multipage/#dom-media-defaultmuted
+    make_bool_setter!(SetDefaultMuted, "muted");
+
     // https://html.spec.whatwg.org/multipage/#dom-media-src
     make_url_getter!(Src, "src");
 
     // https://html.spec.whatwg.org/multipage/#dom-media-src
     make_url_setter!(SetSrc, "src");
 
+    // https://html.spec.whatwg.org/multipage/#dom-media-muted
+    fn Muted(&self) -> bool {
+        self.muted.get() as bool
+    }
+
+    // https://html.spec.whatwg.org/multipage/#dom-media-muted
+    fn SetMuted(&self, value: bool) {
+        self.muted.set(value);
+        let window = window_from_node(self);
+        window
+            .task_manager()
+            .media_element_task_source()
+            .queue_simple_event(self.upcast(), atom!("volumechange"), &window);
```

Comment:
We need to implement the following piece from the [spec](https://html.spec.whatwg.org/multipage/media.html#concept-media-muted) here:

> Then, if the media element is not allowed to play, the user agent must run the internal pause steps for the media element

Reply:
This is also missing from the [SetVolume](https://github.com/servo/servo/blob/master/components/script/dom/htmlmediaelement.rs#L1564) method. It would be really nice if you could also add the check there.